### ISSUE TITLE
ISBN, EAN13, UPC and MPN are not displayed in FO for standard products without combinations

### ISSUE THREAD

**Issue by marionf** [PrestaShop/PrestaShop#15783]
**Describe the bug**
ISBN, EAN13, UPC & MPN are not displayed in FO for standard products

**To Reproduce**
1. Go to Catalog > Products and edit a standard product without combinations
2. Click on Options tab
3. Fill ISBN, EAN13, UPC & MPN fields
4. Go to product page in FO & click on product details tab
5. ISBN, EAN13, UPC & MPN are not displayed while they should be. It's working for a product with combinations.
If these fields are not filled for the selected combination, then you see what is filled in options tab. If fields are filled for the combinations, you should see what is configured for the combinations.

(MPN was added by this PR : #12855) 

**Standard product NOK**

![capture d'écran_2038](https://user-images.githubusercontent.com/13449658/65979018-ed4b8700-e474-11e9-8f24-fce826edd932.png)

![capture d'écran_2039](https://user-images.githubusercontent.com/13449658/65979028-efade100-e474-11e9-8c25-63b90cabb427.png)

**Product with combinations OK**

![capture d'écran_2041](https://user-images.githubusercontent.com/13449658/65979118-0d7b4600-e475-11e9-93fe-e532379fda82.png)

![capture d'écran_2040](https://user-images.githubusercontent.com/13449658/65979126-10763680-e475-11e9-916e-1ee714c9592c.png)

![capture d'écran_2043](https://user-images.githubusercontent.com/13449658/65979148-1835db00-e475-11e9-9aa6-8af154470096.png)

![capture d'écran_2042](https://user-images.githubusercontent.com/13449658/65979153-1a983500-e475-11e9-9fd7-f1f11d7a0af9.png)

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: N/A

**Comment by conscious-creatives** [comment]
Is this still not fixed?? I also have this bug!

**Comment by marionf** [comment]
@conscious-creatives 

Feel free to submint a pull request, if you want this bug fixed more quickly 
https://github.com/PrestaShop/PrestaShop/tree/develop

**Comment by rm-info** [comment]
Couldn't figure out how to fix that correctly (variable $product.specific_references seems to appear from nowhere...), so I did the following:

```smarty
{* if product have specific references, a table will be added to product details section *}
  {block name='product_specific_references'}
  {* KEEP DEFAULT PROCESS ********************************************************* *}
    {if !empty($product.specific_references)}
      <section class="product-features">
        <p class="h6">{l s='Specific References' d='Shop.Theme.Catalog'}</p>
          <dl class="data-sheet">
            {foreach from=$product.specific_references item=reference key=key}
              <dt class="name">{$key}</dt>
              <dd class="value">{$reference}</dd>
            {/foreach}
          </dl>
      </section>
  {* AND ADD THIS TO DEAL WITH SIMPLE PRODUCTS ************************************* *}
	{else}
		{if isset($product.isbn) && !empty($product.isbn)}
			{$product_specific_references['ISBN']=$product.isbn}
		{/if}
		{if isset($product.upc) && !empty($product.upc)}
			{$product_specific_references['UPC']=$product.upc}
		{/if}
		{if isset($product.ean13) && !empty($product.ean13)}
			{$product_specific_references['EAN13']=$product.ean13}
		{/if}
		{if isset($product_specific_references) && !empty($product_specific_references)}
		  <section class="product-features">
			<p class="h6">{l s='Specific References' d='Shop.Theme.Catalog'}</p>
			  <dl class="data-sheet">
				{foreach from=$product_specific_references item=reference key=key}
				  <dt class="name">{$key}</dt>
				  <dd class="value">{$reference}</dd>
				{/foreach}
			  </dl>
		  </section>
		{/if}
    {/if}
  {* END **************************************************************************** *}
  {/block}
```
If someone else has a better idea, I'm all ears.

**Comment by marionf** [comment]
Thanks for diving into this @rm-info 

Would you be willing to make a pull request on GitHub with your code suggestion?
https://github.com/PrestaShop/PrestaShop/tree/develop

This way the prestashop's maintainers will be able to review it and they can suggest changes if necessary

Thank you!

**Comment by rm-info** [comment]
@marionf I'm not used to the pull request thing, but I guess it's proposing a fix to add to the official code.
I'm not sure this would be a good idea since my solution is real crap. But I couldn't figure out how to fix the right way.
Although, as long as we don't have the right answer, this can help.

**Comment by mklimkiew** [comment]
@marionf how about adding this issue to 1.7.7.1 ?
Lack of specific codes are affecting SEO.  
And @rblaurin did a nice PR with mpn and now it seems to be useless :-(

**Comment by marionf** [comment]
Hello @mklimkiew 

This issue can't be part of the 1.7.7.1 as it's not a regression of the 1.7.7.0
You can read in our article about [patch release lifecycle](https://build.prestashop.com/news/ps17-patch-release-lifecycle/), we integrate in the patch versions only regressions from the last minor version or security fixes.

If you know how to fix this issue, you can submit a PR on develop branch to see this issue fixed in 1.7.8.0

**Comment by kmorgen** [comment]
Has this problem been introduced again in a recent update?

I just noticed it is missing again in the product pages (PS 8.2.0).

**Comment by JossVanella** [comment]
Same problem in 8.2.0

**Comment by Angar90** [comment]
> Same problem in 8.2.0

If you want to fix this in PrestaShop 8.2.0, you need to edit the file:
/controllers/front/ProductController.php

and remove in line 1207 this code:
$product['cart_quantity'] = $this->context->cart->getProductQuantity($product['id_product'], $product['id_product_attribute'])['quantity'];

**Comment by hogo20** [comment]
> > Same problem in 8.2.0
> 
> If you want to fix this in PrestaShop 8.2.0, you need to edit the file: /controllers/front/ProductController.php
> 
> and remove in line 1207 this code: $product['cart_quantity'] = $this->context->cart->getProductQuantity($product['id_product'], $product['id_product_attribute'])['quantity'];

hello dear
I made the suggestion you made here. Unfortunately, I saw a new problem. 
When the pre-order is active, already the minimum number of purchases was entered in the relevant box, but now it is not entered automatically. 
Although it is possible to live with this problem, we hope that this problem will be solved in the new version.
